python/tests/kat/t_active_health_check.py

```diff
@@ -0,0 +1,129 @@
+
+        for _ in range(500):
+            yield Query(self.url("healthcheck/"), expected=[200, 500], phase=6)
+            time.sleep(0.06)
+
+    def check(self):
+        # Add up the number of 500 and 200 responses that we got.
+        valid = 0
+        errors = 0
+        for i in range(6, 1006):
+            if self.results[i].status == 200:
+                valid += 1
+            elif self.results[i].status == 500:
+                errors += 1
+
+        # with 1000 requests and 1/5 being an error response, we should have the following distribution +/- 10
+        # assert 190 <= errors <= 210
+        # assert 790 <= valid <= 810
+
+        # But since we configure health chekcing we should actually see 0 errors because the health checks noticed
+        # that one of the pods was unhealthy and didn't route any traffic to it.
+        msg = "Errors: {}, Valid: {}".format(errors, valid)
+        assert errors == 0, msg
+        assert valid == 1000, msg
+
+
+class NoHealthCheckTest(AmbassadorTest):
+    def init(self):
+        self.target = HealthCheckServer()
+
+    def config(self) -> Generator[Union[str, Tuple[Node, str]], None, None]:
+        yield self, self.format(
+            """
+---
+apiVersion: getambassador.io/v3alpha1
+kind: Mapping
+name:  {self.target.path.k8s}-health
+hostname: "*"
+prefix: /healthcheck/
+service: {self.target.path.fqdn}
+resolver: endpoint
+load_balancer:
+  policy: round_robin
+"""
+        )  # The round robin load balancer is not necessary for the test but should help make the request distribution even across the pods
+
+    def queries(self):
+        yield Query(self.url("healthcheck/"), phase=1)  # Just making sure things are running
+        yield Query(self.url("ambassador/v0/diag/"), phase=1)
+
+        yield Query(
+            self.url("healthcheck/makeUnhealthy/"), phase=1
+        )  # the deployment has 5 pods. This will make one of them start returning errors
+
+        # Make 1000 requests and split them up so that we're not hammering the service too much all at once.
+        for _ in range(500):
+            yield Query(self.url("healthcheck/"), expected=[200, 500], phase=2)
+            time.sleep(0.06)
+
+        for _ in range(500):
+            yield Query(self.url("healthcheck/"), expected=[200, 500], phase=3)
+            time.sleep(0.06)
+
+    def check(self):
+        # Since we haven't configured any health checking, we should expect to see a fair number of error responses
+        valid = 0
+        errors = 0
+        for i in range(3, 1003):
+            if self.results[i].status == 200:
+                valid += 1
+            elif self.results[i].status == 500:
+                errors += 1
+        msg = "Errors: {}, Valid: {}".format(errors, valid)
+
+        # with 1000 requests and 1/5 being an error response, we should have the following distribution +/- some
+        assert 175 <= errors <= 225, msg
+        assert 775 <= valid <= 825, msg
```

Comment:
±25 makes me nervous.  In the TCPMapping tests, which do 1000 requests with a 700:300 split, I used a margin of 150.

PS: what I did in `t_tcpmapping.py`, to make the ± margins more readable (IMO):
```python
        # Probabalistic, margin might need tuned
        margin = 150
        assert abs(counts[self.target70.path.k8s] - 700) < margin
        assert abs(counts[self.target30.path.k8s] - 300) < margin
```
(... and pasting that here, I realize that I spelled "Probabilistic" wrong)